src/compiler/types.ts

```diff
@@ -562,6 +562,10 @@ namespace ts {
     export interface ShorthandPropertyAssignment extends ObjectLiteralElement {
         name: Identifier;
         questionToken?: Node;
+        // used when ObjectLiteralExpression is used in ObjectAssignmentPattern
+        // it is grammar error to appear in actual object initializer
+        equalsToken?: Node;

```

Comment:
the comment is referring to both `equalsToken` and `objectAssignmentInitializer` - this pair matches `Initializer` part from `CoverInitializerName` [production](https://people.mozilla.org/~jorendorff/es6-draft.html#sec-object-initializer). My reasoning of reporting error on `=` instead of entire range `[equalsToken, objectAssignmentInitializer]` was:
- having large ranges of squigglies is annoying
- I can certainly see cases when by mistake people type `{x = 1}` instead of `{x: 1}` and personally for me it will look confusing if we squiggle `= 1` instead of just =`